diff --git a/gpu/amd_linux.go b/gpu/amd_linux.go
index cbbdf03..403d176 100644
--- a/gpu/amd_linux.go
+++ b/gpu/amd_linux.go
@@ -18,7 +18,7 @@ import (
 // Discovery logic for AMD/ROCm GPUs
 
 const (
-	DriverVersionFile     = "/sys/module/amdgpu/version"
+	DriverVersionFile     = "/usr/share/ollama/gentoo_amdgpu_version"
 	AMDNodesSysfsDir      = "/sys/class/kfd/kfd/topology/nodes/"
 	GPUPropertiesFileGlob = AMDNodesSysfsDir + "*/properties"
 
@@ -28,10 +28,8 @@ const (
 	RocmStandardLocation   = "/opt/rocm/lib"
 )
 
-var (
-	// Used to validate if the given ROCm lib is usable
-	ROCmLibGlobs = []string{"libhipblas.so.2*", "rocblas"} // TODO - probably include more coverage of files here...
-)
+// Used to validate if the given ROCm lib is usable
+var ROCmLibGlobs = []string{"libhipblas.so.2*", "rocblas"} // TODO - probably include more coverage of files here...
 
 // Gather GPU information from the amdgpu driver if any supported GPUs are detected
 func AMDGetGPUInfo() []GpuInfo {
@@ -312,7 +310,7 @@ func AMDValidateLibDir() (string, error) {
 	}
 
 	// Well known ollama installer path
-	installedRocmDir := "/usr/share/ollama/lib/rocm"
+	installedRocmDir := "/lib64"
 	if rocmLibUsable(installedRocmDir) {
 		return installedRocmDir, nil
 	}
