diff --git a/rocclr/device/rocm/rocdevice.cpp b/rocclr/device/rocm/rocdevice.cpp
index 757a113fe..6bce3b59d 100644
--- a/rocclr/device/rocm/rocdevice.cpp
+++ b/rocclr/device/rocm/rocdevice.cpp
@@ -2342,10 +2342,10 @@ void* Device::deviceLocalAlloc(size_t size, bool atomics, bool pseudo_fine_grain
     return nullptr;
   }
 
-  uint32_t hsa_mem_flags = 0;
-  if (contiguous) {
-    hsa_mem_flags = HSA_AMD_MEMORY_POOL_CONTIGUOUS_FLAG;
-  }
+  //uint32_t hsa_mem_flags = 0;
+  //if (contiguous) {
+  //  hsa_mem_flags = HSA_AMD_MEMORY_POOL_CONTIGUOUS_FLAG;
+  //}
 
   void* ptr = nullptr;
   hsa_status_t stat = hsa_amd_memory_pool_allocate(pool, size, 0, &ptr);
